collect-artifacts.js

```diff
@@ -1,18 +1,15 @@
-var fs = require('fs')
-var path = require('path')
 var error = require('./error')
+var recursive = require('recursive-readdir')
 
 module.exports = collectArtifacts
 
 function collectArtifacts (release, opts, cb) {
   var fileExp = opts['include-regex']
-  fs.readdir(release, function (err, files) {
+  recursive(release, function (err, files) {
     if (err) return cb(err)
 
     var collected = files.filter(function filterByRegex (filename) {
-      return fileExp.test(filename)
-    }).map(function addPath (filename) {
-      return path.join(release, filename)
+      return fileExp.test(filename.substring(release.length + 1))
```

Comment:
Bear with me.. This will not work in cases where people use a custom release folder (through `binary.module_path` in `package.json`) and it happens to be something (e.g. dots) that `path.join()` (called in `recursive-readdir`) normalizes to a shorter string. That's an edge case, admittedly, but cases like that are why it's generally preferred to use `path` functions rather than "manual" string slicing.

Reply:
Maybe we can use a different package for recursive reading, one that does not prefix the results. Then the filtering and mapping code here can stay the same.

Reply:
What do you think about using path.relative(), to get the filename w/o prefix?  Like so :
```js
      return fileExp.test(path.relative(release, filename))
```
It will resolve both to absolute path(according to the doc) before returning the relative part, so it will behave exactly like removing `release`, no matter what kind of path is used for it.

Reply:
Sure. That should work.